File: iosApp/iosApp/ui/screens/movie/movies/MoviesState.swift
//
//  MoviesState.swift
//  iosApp
//
//  Created by ahmed elbagory on 11/04/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import shared
struct MoviesState {
    var loading: Bool
    var movies: [MovieItem]
    var errorMessage: String
    var loadFinished: Bool
    
    init(loading: Bool = false, movies: [MovieItem] = [], errorMessage: String = "", loadFinished: Bool = false) {
        self.loading = loading
        self.movies = movies
        self.errorMessage = errorMessage
        self.loadFinished = loadFinished
    }
}

File: androidApp/src/main/java/kmm/superApp/android/ui/screens/movie/movies/MoviesState.kt
package kmm.superApp.android.ui.screens.movie.movies

import kmm.superApp.domain.entity.MovieItem


data class MoviesState (
    var loading: Boolean = false,
    var movies: List<MovieItem> = listOf(),
    var errorMessage: String = "",
    var loadFinished: Boolean = false
)